### ISSUE TITLE
Error parsing GitVersion output on Teamcity Linux agent when running task

### ISSUE THREAD

**Issue by robertmircea** [nuke-build/nuke#367]
### Description

`Error parsing Infinity value. Path '', line 1, position 1` while parsing gitversion's output when running nuke task on Teamcity Linux agent. See stacktrace below.

### Steps to reproduce

1. Building on Teamcity with .teamcity Kotlin project generated by Nuke
```
[TeamCity(TeamCityAgentPlatform.Unix, 
    ManuallyTriggeredTargets = new[] {nameof(PublishLinux), nameof(PublishWin10)},
    NonEntryTargets = new[] { nameof(Restore) },
    ExcludedTargets = new[] { nameof(Clean) },
    VcsTriggeredTargets = new []{ nameof(Compile)},
    DefaultBranch = "master",
    AutoGenerate = false
)]
class Build : NukeBuild
[...]
```

Running the following task:
```
object Compile : BuildType({
    name = "Compile"
    vcs {
        root(VcsRoot)
        cleanCheckout = true
    }
    steps {
        exec {
            path = "build.sh"
            arguments = "Restore Compile --skip"
        }
    }
    triggers {
        vcs {
            branchFilter = ""
            triggerRules = "+:**"
        }
    }
})
```

2. Using GitVersion 5.1.1 and latest pre-release Nuke 0.24.0-alpha0193

Excerpt from `_build.csproj`:
```
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.0</TargetFramework>
    <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
    <RootNamespace></RootNamespace>
    <IsPackable>False</IsPackable>
    <NoWarn>CS0649;CS0169</NoWarn>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Nuke.Common" Version="0.24.0-alpha0193" />
    <PackageDownload Include="GitVersion.Tool" Version="[5.1.1]" />
  </ItemGroup>
```
3. When running the build on Linux Teamcity agent, I receive the following error when Nuke parses GitVersion output and build fails

### Relevant log output

```
Step 1/1: Command Line
  Starting: /srv/teamcity-agents/teamcity-agent-1.2/work/df8b84bc28b01fff/build.sh Restore Compile --skip
  in directory: /srv/teamcity-agents/teamcity-agent-1.2/work/df8b84bc28b01fff
  GNU bash, version 4.3.30(1)-release (x86_64-pc-linux-gnu)
  dotnet_install: Warning: Unable to locate zlib. Probable prerequisite missing; install zlib.
  dotnet_install: Warning: Unable to locate liblttng. Probable prerequisite missing; install libcurl.
  dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Sdk/3.1.100/dotnet-sdk-3.1.100-linux-x64.tar.gz
  dotnet-install: Extracting zip from https://dotnetcli.azureedge.net/dotnet/Sdk/3.1.100/dotnet-sdk-3.1.100-linux-x64.tar.gz
  dotnet-install: Binaries of dotnet can be found in /srv/teamcity-agents/teamcity-agent-1.2/work/df8b84bc28b01fff/.tmp/dotnet-unix
  dotnet-install: Installation finished successfully.
  Microsoft (R) .NET Core SDK version 3.1.100
  Microsoft (R) Build Engine version 16.4.0+e901037fe for .NET Core
  Copyright (C) Microsoft Corporation. All rights reserved.
  
    Restore completed in 2.5 sec for /srv/teamcity-agents/teamcity-agent-1.2/work/df8b84bc28b01fff/build/_build.csproj.
    _build -> /srv/teamcity-agents/teamcity-agent-1.2/work/df8b84bc28b01fff/build/bin/Release/_build.dll
  
  Build succeeded.
      0 Warning(s)
      0 Error(s)
  
  Time Elapsed 00:00:07.28
  
  ███╗   ██╗██╗   ██╗██╗  ██╗███████╗
  ████╗  ██║██║   ██║██║ ██╔╝██╔════╝
  ██╔██╗ ██║██║   ██║█████╔╝ █████╗  
  ██║╚██╗██║██║   ██║██╔═██╗ ██╔══╝  
  ██║ ╚████║╚██████╔╝██║  ██╗███████╗
  ╚═╝  ╚═══╝ ╚═════╝ ╚═╝  ╚═╝╚══════╝
  
  NUKE Execution Engine version 0.24.0-alpha0193 (Linux,.NETCoreApp,Version=v3.0)
  
  > /srv/teamcity-agents/teamcity-agent-1.2/work/df8b84bc28b01fff/.tmp/dotnet-unix/dotnet /home/teamcity/.nuget/packages/gitversion.tool/5.1.1/tools/netcoreapp3.0/any/GitVersion.dll
  Error parsing Infinity value. Path '', line 1, position 1.
     at Newtonsoft.Json.JsonTextReader.ParseNumberPositiveInfinity(ReadType readType, Boolean matched)
     at Newtonsoft.Json.JsonTextReader.ParseValue()
     at Newtonsoft.Json.JsonReader.ReadAndMoveToContent()
     at Newtonsoft.Json.JsonReader.ReadForType(JsonContract contract, Boolean hasConverter)
     at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
     at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
     at Newtonsoft.Json.JsonSerializer.Deserialize(JsonReader reader, Type objectType)
     at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
     at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)
     at Nuke.Common.Tools.GitVersion.GitVersionTasks.GetResult(IProcess process, GitVersionSettings toolSettings)
     at Nuke.Common.Tools.GitVersion.GitVersionTasks.GitVersion(GitVersionSettings toolSettings)
     at Nuke.Common.Tools.GitVersion.GitVersionTasks.GitVersion(Configure`1 configurator)
     at Nuke.Common.Tools.GitVersion.GitVersionAttribute.GetValue(MemberInfo member, Object instance)
     at Nuke.Common.Execution.InjectionUtility.InjectValuesInternal[T](T instance, IEnumerable`1 tuples)
```

**Comment by matkoch** [comment]
@robertmircea starring is required ;)

**Comment by matkoch** [comment]
See https://github.com/GitTools/GitVersion/issues/1982

**Comment by robertmircea** [comment]
Which is the conclusion? Is it a bug and it will be fixed in 0.24.0? Sorry but is not clear for me from Gitversion#1982

**Comment by matkoch** [comment]
We can’t fix it, just provide a better error message.
